node.js

```diff
@@ -400,64 +493,67 @@ async function join(known_node) {
         predecessor = _self;
     }
 
-    // TODO move keys: (predecessor, n]; i.e., (predecessor, _self]
+    // TODO migrate keys: (predecessor, n]; i.e., (predecessor, _self]
+    await migrate_keys();
+
+    // initialize successor table - deviates from SIGCOMM
+    SuccessorTable[0] = FingerTable[0].successor;
 
-    /* vvvvv     vvvvv     TBD debugging code     vvvvv     vvvvv */
     if (DEBUGGING_LOCAL) {
         console.log(">>>>>     join          ");
-        console.log("The FingerTable[] leaving join() is:\n", FingerTable);
-        console.log("The predecessor leaving join() is ", predecessor);
+        console.log("The FingerTable[] leaving {", _self.id, "}.join(", known_node.id, ") is:\n", FingerTable);
```

Comment:
Please look at the syntax for template strings. The syntax is much cleaner for this sort of thing. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals